### PR TITLE
feat: generating a URL based on dimension columns

### PR DESCRIPTION
**Pull request by tpluscode**
related to #34

Fact table no also have a identifier template thus becoming almost the same as dimension tables.
The only difference is that a tact table does not require the template and will then generate blank node ids

### CODE REVIEW

**Comment by martinmaillard** [comment]
Is there some SPARQL migration queries that must be done for that to work with existing projects?

**Comment by tpluscode** [comment]
No

**Comment by martinmaillard** [comment]
What if the combination of all dimension tables is not unique?

For example in the BLW dataset, I defined the "calendar week" dimension as an attribute of the facts table because I don't want a URI for each week. I want it to be a literal with type `xsd:date`. But if you leave it out for the observation identifier, there will be clashes.

**Comment by tpluscode** [comment]
> What if the combination of all dimension tables is not unique?

@ktk

File: packages/rdfine-data-cube/Table/Table.ts [hunk 1]
```diff
@@ -88,7 +89,7 @@ export function FactTableMixin<TBase extends Constructor> (Base: TBase) {
       }
 
       const path = [
-        this.name,
+        urlSlug(this.name),
```
Comment (martinmaillard) [inline packages/rdfine-data-cube/Table/Table.ts]: Should this fix also be applied to the `aboutUrl` property of the mapping?

Comment (tpluscode) [reply packages/rdfine-data-cube/Table/Table.ts]: Could you be more specific?

Comment (martinmaillard) [reply packages/rdfine-data-cube/Table/Table.ts]: Forget it, I think I'm confusing stuff.

**Pull request merged**